src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/ContactsController.php

```diff
@@ -105,43 +110,92 @@ public function searchAction(Request $request)
     {
-        $legacyLink = $this->getAdminLink('AdminContacts', [
-            'addcontact' => 1,
-        ]);
+        $contactFormBuilder = $this->get('prestashop.core.form.identifiable_object.builder.contact_form_builder');
+        $contactForm = $contactFormBuilder->getForm();
+        $contactForm->handleRequest($request);
+
+        try {
+            $contactFormHandler = $this->get('prestashop.core.form.identifiable_object.handler.contact_form_handler');
+            $result = $contactFormHandler->handle($contactForm);
+
+            if ($result->isSubmitted() && $result->isValid()) {
+                $this->addFlash(
+                    'success',
+                    $this->trans('Successful creation.', 'Admin.Notifications.Success')
+                );
+
+                return $this->redirectToRoute('admin_contacts_index');
+            }
+        } catch (DomainException $exception) {
+            $this->addFlash('error', $this->handleException($exception));
+        }
 
-        return $this->redirect($legacyLink);
+        return $this->render('@PrestaShop/Admin/Configure/ShopParameters/Contact/Contacts/add.html.twig', [
+            'contactForm' => $contactForm->createView(),
+        ]);
     }
 
     /**
      * Display the contact edit form.
      *
-     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))", message="You do not have permission to edit this.")
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_contacts_index",
+     *     message="You do not have permission to edit this."
+     * )
      *
      * @param int $contactId
+     * @param Request $request
      *
-     * @return RedirectResponse
+     * @return Response
      */
-    public function editAction($contactId)
+    public function editAction($contactId, Request $request)
     {
-        $legacyLink = $this->getAdminLink('AdminContacts', [
-            'id_contact' => $contactId,
-            'updatecontact' => 1,
-        ]);
+        $contactFormBuilder = $this->get('prestashop.core.form.identifiable_object.builder.contact_form_builder');
+        $contactForm = $contactFormBuilder->getFormFor((int) $contactId);
+
+        $contactForm->handleRequest($request);
+
+        try {
+            $contactFormHandler = $this->get('prestashop.core.form.identifiable_object.handler.contact_form_handler');
+            $result = $contactFormHandler->handleFor((int) $contactId, $contactForm);
 
-        return $this->redirect($legacyLink);
+            if (null !== $result->getIdentifiableObjectId()) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_contacts_index');
+            }
+        } catch (DomainException $exception) {
+            $this->addFlash('error', $this->handleException($exception));
```

Comment:
would you mind using `$this->getErrorMessageForException` for exception handling? you can refer to `LanguageController` for example.

Reply:
well yes but the usage in `ContactsController` is very similar - don't see a strong argument to change it